lib/orbit_web/controllers/sign_in_controller.ex

```diff
@@ -31,16 +31,28 @@ defmodule OrbitWeb.SignInController do
                 ^start_datetime <= si.signed_in_at and
                   si.signed_in_at < ^end_datetime and
                   si.rail_line == ^rail_line,
-              preload: [:signed_in_employee, :signed_in_by_user],
-              order_by: [desc: :signed_in_at]
+              join: u in assoc(si, :signed_in_by_user),
+              preload: [:signed_in_employee, signed_in_by_user: u],
```

Comment:
**suggestion (non-blocking):** If you're already in here turning the preload for `signed_in_by_user` from the default (which uses a separate query - see [here](https://hexdocs.pm/ecto/Ecto.Query.html#preload/3)) to using an explicit join in the same query, it might be worth doing the same for `signed_in_employee`, thus turning this whole operation into a single query.

Reply:
ah! thanks, I see you did that in https://github.com/mbta/orbit/pull/103. why is that not the default preload behavior, though? I guess because Ecto wants joins to remain explicit?

Reply:
f8ad95b

Reply:
> why is that not the default preload behavior, though?

I'm not sure - it's also possible that Ecto isn't able to generate the join statements fully programmatically or something? In any event it's just an interesting quirk to be aware of.